uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/AbsencePeriod.php

```diff
@@ -104,6 +106,34 @@ private static function validateDates($params) {
   }
 
   /**
+   * Checks if another absence period exists with same title as
+   * the absence period being created/updated and throws an exception if found.
+   *
+   * @param array $params
+   *
+   * @throws \CRM_HRLeaveAndAbsences_Exception_InvalidAbsencePeriodException
+   */
+  private static function validateAbsencePeriodTitle($params) {
+    $title = !empty($params['title']) ? $params['title'] : '';
+    $absencePeriod = new self();
```

Comment:
This will be faster than the API. Will eliminate a lot of checks that the API has to make before the actual query to the db.
Besides, API will make more sense if the data to be retrieved is in another extension but since its within the same extension.